### DISCUSSION TITLE
High CPU usage and network traffic with Ejabberd 22.05

### DISCUSSION THREAD

**Discussion by danycloutier** [processone/ejabberd#4183] [Questions]
Hi,
We maintain an ejabberd cluster of 4 nodes and we recently upgraded from 21.12 to 22.05. Since then, we see a lot more cpu usage and network traffic.
We tested with both erlang 22 and 23 with the same results and we run on Debian Buster.

Network traffic:
![image](https://github.com/processone/ejabberd/assets/63265780/49ec28ac-9f2a-4521-94c5-e497447c76c3)

CPU usage:
![image](https://github.com/processone/ejabberd/assets/63265780/43f030f3-87be-4c0f-8175-106b62708aac)

Looking with htop, I can see there's cpu spikes like this every minute
![image](https://github.com/processone/ejabberd/assets/63265780/2c515072-0f1c-41d7-a934-17e541bebf76)

The spikes seems to happen every 60 seconds. I have a timer that trigger each minutes to make a check in a database, but I deactivated this custom module and I don't see any changes.

Do you have an idea of what could cause this in the 22.05 release ? Any help would be appreciated :) 

Merci !

**Comment by licaon-kter** [comment]
> Debian Buster

So you build your own ejabberd or why stay on 22.05 when latest is 24.02?

**Reply by licaon-kter** [reply]
`ejabberdctl etop` says what?

**Comment by prefiks** [comment]
Do you have any error in your logs?

**Comment by danycloutier** [comment]
It turns out to be caused by [this](https://github.com/processone/ejabberd/commit/8b7da70b57096d5c88c4128877e8802a12057bd4#diff-5189d65bb6c76e6c8e0b2d15813b1ba12ab2385e831317be46c062d991ebeb51R1130). It generates a lot of messages on a remove_user.

**Comment by prefiks** [comment]
I am guessing that from muc room cleanup code? Do you have lots of rooms active on that server?

**Comment by Neustradamus** [comment]
@danycloutier: Have you upgraded to 24.02 to see difference?
